contracts/test/veLON/vBalance.t.sol

```diff
@@ -189,6 +189,75 @@ contract TestVeLONBalance is TestVeLON {
+    function testVBalanceOfAtBlkBeforeLastPoint() public {
+        uint256 aliceTokenId = _stakeAndValidate(user, DEFAULT_STAKE_AMOUNT, 2 weeks);
+        uint256 aliceBalance = _initialvBalance(DEFAULT_STAKE_AMOUNT, 2 weeks);
+
+        assertEq(veLon.epoch(), 1);
+        assertEq(veLon.vBalanceOfAtBlk(aliceTokenId, block.number), aliceBalance);
+
+        // fast forward 1 week and second user createlock
+        vm.warp(block.timestamp + 1 weeks);
+        vm.roll(block.number + 1);
+        uint256 bobTokenId = _stakeAndValidate(bob, DEFAULT_STAKE_AMOUNT, 2 weeks);
+        uint256 bobBalance = _initialvBalance(DEFAULT_STAKE_AMOUNT, 2 weeks);
+
+        // verify the vBalanceOfAtblk of 1st block when Alcie createLock
+        assertEq(veLon.vBalanceOfAtBlk(aliceTokenId, block.number - 1), aliceBalance);
+
+        // Continue fast forward 1 week
+        // Bob extend lock for 2 weeks
+        vm.warp(block.timestamp + 1 weeks);
+        vm.roll(block.number + 1);
+        vm.prank(bob);
+        veLon.extendLock(bobTokenId, 2 weeks);
+
+        // Verify the vBalanceOfAtBlk when first block Alice createLock
+        assertEq(veLon.vBalanceOfAtBlk(aliceTokenId, block.number - 2), aliceBalance);
+    }
+
+    // test vBalanceOfAtTime() when the target _t is before lastPoint.ts
+    function testVBalanceOfAtTimeBeforeLastPoint() public {
+        uint256 tokenId = _stakeAndValidate(user, DEFAULT_STAKE_AMOUNT, 2 weeks);
+        uint256 expectedBalance = _initialvBalance(DEFAULT_STAKE_AMOUNT, 2 weeks);
+
+        assertEq(veLon.userPointEpoch(1), 1);
+        assertEq(veLon.vBalanceOfAtTime(tokenId, block.timestamp), expectedBalance);
+
+        // fast forward 1 weeks and user extend lock for 3 weeks, so the the vBalance is 2 week more
+        // The userEpoch should be 2 (createLock, extendLock)
+        // And the lastPoint of user should be updated now becase user extendedLock
+        vm.warp(block.timestamp + 1 weeks);
+        vm.roll(block.number + 1);
+        vm.prank(user);
+        veLon.extendLock(tokenId, 3 weeks);
+
+        // expetedBalance  = initialBalance - 1 weeks passed + 2 week extended
+        assertEq(veLon.userPointEpoch(tokenId), 2);
+        expectedBalance = (expectedBalance - (10 * 1 weeks) + (10 * 2 weeks));
+        assertEq(veLon.vBalanceOf(tokenId), expectedBalance);
+        assertEq(veLon.userPointEpoch(tokenId), 2);
+
+        // Continue fast forward 1 week, so now is the third week(2 weeks passed since createLock)
+        // Query the vBalanceOfAtime in first week (the time lock was created)
+        // The traget time is before lastPoint.ts(the time lock was extended)
+        vm.warp(block.timestamp + 1 weeks);
+        vm.roll(block.number + 1);
+        uint256 initialBalance = _initialvBalance(DEFAULT_STAKE_AMOUNT, 2 weeks);
+        assertEq(veLon.userPointEpoch(tokenId), 2);
+        assertEq(veLon.vBalanceOfAtTime(tokenId, block.timestamp - 2 weeks), initialBalance);
```

Comment:
How about check balance at `ts` other than the begining? For example a `ts` between 2 user points.

Reply:
change in [ffd08d6](https://github.com/consenlabs/tokenlon-contracts/pull/81/commits/ffd08d65e0fe8a5308e3b267720d93e4543d6c45)

Reply:
change in https://github.com/consenlabs/tokenlon-contracts/pull/81/commits/ffd08d65e0fe8a5308e3b267720d93e4543d6c45